superset-frontend/src/explore/components/controls/ControlPopover/ControlPopover.tsx

```diff
@@ -89,24 +98,50 @@ const ControlPopover: React.FC<PopoverProps> = ({
 
   const handleOnVisibleChange = useCallback(
     (visible: boolean) => {
-      if (props.visible === undefined) {
+      if (visible === undefined) {
         changeContainerScrollStatus(visible);
       }
 
+      setVisible(!!visible);
       props.onVisibleChange?.(visible);
     },
     [props, changeContainerScrollStatus],
   );
 
+  const handleDocumentKeyDownListener = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setVisible(false);
+        props.onVisibleChange?.(false);
```

Comment:
`props.onVisibleChange?.(false)` is enough. Why did you introduce `visible` state and `setVisible`?

Reply:
To control the popover state in this component, to enable closing on escape.